cmd/registry/main.go

```diff
@@ -17,8 +18,34 @@ import (
 	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
 	_ "github.com/docker/distribution/registry/storage/driver/s3-goamz"
 	_ "github.com/docker/distribution/registry/storage/driver/swift"
+	"github.com/spf13/cobra"
 )
 
+var subCommands = map[string]*cobra.Command{
+	registry.GCCmd.Name(): registry.GCCmd,
+}
+
+func findSubcommand(subCommands map[string]*cobra.Command, args []string) *cobra.Command {
+	if len(args) < 2 {
+		return nil
+	}
+
+	potentialSubCommand := args[1]
+	if command, ok := subCommands[potentialSubCommand]; ok {
+		return command
+	}
+
+	return nil
+}
+
 func main() {
+	// check if first arg is a subcommand. If not, assume args are for the main
+	// command that starts up a registry server
+	subcommand := findSubcommand(subCommands, os.Args)
```

Comment:
I'm happy with either way, but just to clarify what we need.  Either,
1. A bash script called (`bin/registry`) to ensure `bin/registry` <config> calls `bin/registry serve` or
2. An explicit `serve` subcommand and updating the project and official image Dockerfile with a default entrypoint of `bin/registry serve`

2 is simpler, though I would like the project Dockerfile change with this PR (I can advise @andrewnguyen and take care of the official image dockerfile).